packages/ember-htmlbars/tests/integration/component_invocation_test.js

```diff
@@ -613,6 +613,47 @@ QUnit.test("components in template of a yielding component should have the prope
   equal(outer.parentView, view, 'x-outer receives the ambient scope as its parentView');
 });
 
+QUnit.test("components should receive the viewRegistry from the parent view", function() {
+  var outer, innerTemplate, innerLayout;
+
+  var viewRegistry = {};
+
+  registry.register('component:x-outer', Component.extend({
+    init() {
+      this._super(...arguments);
+      outer = this;
+    }
+  }));
+
+  registry.register('component:x-inner-in-template', Component.extend({
+    init() {
+      this._super(...arguments);
+      innerTemplate = this;
+    }
+  }));
+
+  registry.register('component:x-inner-in-layout', Component.extend({
+    init() {
+      this._super(...arguments);
+      innerLayout = this;
+    }
+  }));
+
+  registry.register('template:components/x-outer', compile('{{x-inner-in-layout}}{{yield}}'));
+
+  view = EmberView.extend({
+    _viewRegistry: viewRegistry,
+    template: compile('{{#x-outer}}{{x-inner-in-template}}{{/x-outer}}'),
+    container: container
+  }).create();
+
+  runAppend(view);
+
+  equal(innerTemplate._viewRegistry, viewRegistry);
```

Comment:
By default `_viewRegistry` is `Ember.View.views`, but that does not work in the fastboot or multiple app scenario (because there are two different view registries).

---

This change, ensures that the `_viewRegistry` is passed down through the various views/components created during render (it was previously not passed through).  It does not actually test that things use `_viewRegistry` (though they should).